src/EditorFeatures/Test/CodeFixes/CodeFixServiceTests.cs

```diff
@@ -768,5 +797,132 @@ private void RegisterCodeFix(CodeFixContext context, string titleSuffix)
+        {
+            var analyzer = new AdditionalFileAnalyzer();
+            var fixer1 = new AdditionalFileFixerWithDocumentKindsAndExtensions();
+            var fixer2 = new AdditionalFileFixerWithDocumentKinds();
+            var fixer3 = new AdditionalFileFixerWithDocumentExtensions();
+            var fixer4 = new AdditionalFileFixerWithoutDocumentKindsAndExtensions();
+            var fixers = ImmutableArray.Create<CodeFixProvider>(fixer1, fixer2, fixer3, fixer4);
+            var analyzerReference = new MockAnalyzerReference(fixers, ImmutableArray.Create<DiagnosticAnalyzer>(analyzer));
+
+            // Verify available code fixes for .txt additional document
+            var tuple = ServiceSetup(fixers, additionalDocument: new TestHostDocument("Additional Document", filePath: "test.txt"));
+            using var workspace = tuple.workspace;
+            GetDocumentAndExtensionManager(tuple.analyzerService, workspace, out var txtDocument, out var extensionManager, analyzerReference, documentKind: TextDocumentKind.AdditionalDocument);
+            var txtDocumentCodeFixes = await tuple.codeFixService.GetFixesAsync(txtDocument, TextSpan.FromBounds(0, 1), CodeActionOptions.DefaultProvider, isBlocking: false, CancellationToken.None);
+            Assert.Equal(2, txtDocumentCodeFixes.Length);
+            var txtDocumentCodeFixTitles = txtDocumentCodeFixes.Select(s => s.Fixes.Single().Action.Title).ToImmutableArray();
+            Assert.Contains(fixer1.Title, txtDocumentCodeFixTitles);
+            Assert.Contains(fixer2.Title, txtDocumentCodeFixTitles);
+
+            // Verify code fix application
+            var codeAction = txtDocumentCodeFixes.Single(s => s.Fixes.Single().Action.Title == fixer1.Title).Fixes.Single().Action;
+            var solution = await codeAction.GetChangedSolutionInternalAsync();
+            var changedtxtDocument = solution!.Projects.Single().AdditionalDocuments.Single(t => t.Id == txtDocument.Id);
+            Assert.Equal("Additional Document", txtDocument.GetTextSynchronously(CancellationToken.None).ToString());
+            Assert.Equal($"Additional Document{fixer1.Title}", changedtxtDocument.GetTextSynchronously(CancellationToken.None).ToString());
+
+            // Verify available code fixes for .log additional document
+            tuple = ServiceSetup(fixers, additionalDocument: new TestHostDocument("Additional Document", filePath: "test.log"));
+            using var workspace2 = tuple.workspace;
+            GetDocumentAndExtensionManager(tuple.analyzerService, workspace2, out var logDocument, out extensionManager, analyzerReference, documentKind: TextDocumentKind.AdditionalDocument);
+            var logDocumentCodeFixes = await tuple.codeFixService.GetFixesAsync(logDocument, TextSpan.FromBounds(0, 1), CodeActionOptions.DefaultProvider, isBlocking: false, CancellationToken.None);
+            Assert.Equal(1, logDocumentCodeFixes.Length);
```

Comment:
Assert.Single gives a better error than this, and also returns the object so you can skip the .Single() in the next line.